cinema/views.py

```diff
@@ -127,6 +128,29 @@ def upload_image(self, request, pk=None):
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
```

Comment:
For the upload_image method, consider handling the case where the pk parameter is None. As it stands, if pk is None, the get_object method will raise a 404 error. It might be more user-friendly to catch this error and return a custom error message.